rest-api-spec/test/cluster.state/40_allow_no_indices.yaml

```diff
@@ -0,0 +1,18 @@
+---
+"Test allow_no_indices parameter":
+
+  - do:
+      cluster.state:
+        metric: [ metadata ]
+        index: test*
+
+  - match: {metadata.indices: {}}
+  
+  - do:
+      catch: missing
+      cluster.state:
+        metric: [ metadata ]
+        index: test*
+        allow_no_indices: false
+
+  - is_false: metadata
```

Comment:
I think this assert doesn't make much sense given that the last call returns an error. `catch: missing` seems enough.